src/prototype_room_init.js

```diff
@@ -63,6 +63,42 @@ Room.prototype.setTowerFiller = function() {
   }
 };
 
+function setLabsTerminal(room, path, costMatrixBase) {
+  for (let pathI = path.length - 1; pathI > 0; pathI--) {
+    let pathPos = new RoomPosition(path[pathI].x, path[pathI].y, room.name);
+    let structurePosIterator = pathPos.findNearPosition();
+    for (let structurePos of structurePosIterator) {
+      if (room.memory.position.structure.lab.length < CONTROLLER_STRUCTURES.lab[8]) {
+        room.memory.position.structure.lab.push(structurePos);
+        costMatrixBase.set(structurePos.x, structurePos.y, config.layout.structureAvoid);
+        continue;
+      }
+      if (room.memory.position.structure.terminal.length < CONTROLLER_STRUCTURES.terminal[8]) {
```

Comment:
I think it is better to get the mineral out of the extractor faster. If the terminal is close, the extractor creep would be more efficient.

Distribution can be optimized with more or other creeps.